src/Sets/Interval.jl

```diff
@@ -514,3 +515,24 @@ We consider the interval as a line segment with y coordinate equal to zero.
 function plot_recipe(I::Interval{N}, ε::N=zero(N)) where {N<:Real}
     return [min(I), max(I)], zeros(N, 2)
 end
+
+"""
+    linear_map(M::AbstractMatrix{N}, x::Interval{N}) where {N<:Real}
+
+Concrete linear map of an interval.
+
+### Input
+
+- `M` -- matrix
+- `x` -- interval
+
+### Output
+
+An interval obtained by scaling `x` by the matrix `M`.
+"""
+function linear_map(M::AbstractMatrix{N}, x::Interval{N}) where {N<:Real}
+    @assert size(M) == (1, 1) "a linear map of size $(size(M)) " *
```

Comment:
Do we always require that `M` is square? I think you should be able to have a different output dimension in principle.

Reply:
> Do we always require that M is square?

No, because there can be projections. An interval has dimension 1 so it cannot be projected over a set of dimension smaller than 1 -- and it is not clear what does `M_{2x1} * Interval` represent. Anyhow, I can change the assertion to the weaker `size(M, 2) == 1` if you prefer.

Reply:
> and it is not clear what does M_{2x1} * Interval represent

In `master` it works just fine:
```julia
julia> M = hcat([1., 2., 3.])
3×1 Array{Float64,2}:
 1.0
 2.0
 3.0

julia> X = LazySets.Interval(0., 1.);

julia> linear_map(M, X)
Zonotope{Float64}([0.5, 1.0, 1.5], [0.5; 1.0; 1.5])

julia> P = convert(HPolyhedron, X)
HPolyhedron{Float64}(HalfSpace{Float64,VN} where VN<:AbstractArray{Float64,1}[HalfSpace{Float64,LazySets.Arrays.SingleEntryVector{Float64}}([1.0], 1.0), HalfSpace{Float64,LazySets.Arrays.SingleEntryVector{Float64}}([-1.0], -0.0)])

julia> linear_map(M, P)
VPolytope{Float64}(Array{Float64,1}[[1.0, 2.0, 3.0], [0.0, 0.0, 0.0]])
```
So I suggest to (a) weaken the constraint and (b) make it work for `size(M, 1) > 1` as well. We can still return an `Interval` if `size(M, 1) == 1` of course.

Reply:
Ok, i agree with your idea. The drawback is that `linear_map(::AbstractMatrix, ::Interval)` is now type unstable, since the dimension is not available at runtime.

Reply:
> The drawback is that linear_map(::AbstractMatrix, ::Interval) is now type unstable, since the dimension is not available at compile time.

Yes, so maybe you want to oursource the `Interval` case to a new function `linear_map_1d` which you could also call directly instead.